legend-engine-ide-lsp-default-extensions/src/test/java/org/finos/legend/engine/ide/lsp/extension/notebook/TestPureBookLSPGrammarExtension.java

```diff
@@ -291,51 +270,39 @@ void completions()
-                        ")");
+        MutableMap<String, String> codeFiles = this.getCodeFilesThatParseCompile();
+        stateForTestFactory.newSectionStates(gs, codeFiles);
+        Map<String, String> executableArgs = Map.of(
+                "code", "->extend(~col_e4751a0c_6b6d_49ff_a929_2b2e0d13ff51:x|$x.",
+                "baseQuery", "{\"_type\":\"lambda"
+                        + "\",\"body\":[{\"_type\":\"func\",\"function\":\"from\"," +
+                        "\"parameters\":[{\"_type\":\"func\"," + "\"function\":\"select\"," +
+                        "\"parameters\":[{\"_type\":\"classInstance\"," + "\"multiplicity\":{\"lowerBound\":1," +
+                        "\"upperBound\":1},\"type\":\">\"," + "\"value\":{\"path\":[\"test::h2Store\"," +
+                        "\"personTable\"]}}]},{\"_type\":\"packageableElementPtr\"," + "\"fullPath\":\"test" +
+                        "::h2Runtime\"}]}],\"parameters\":[]}"
+        );
 
-        stateForTestFactory.newSectionState(gs, "connection.pure",
-                "###Connection\n" +
-                        "RelationalDatabaseConnection test::h2Conn\n" +
-                        "{\n" +
-                        "    store: test::h2Store; \n" +
-                        "    type: H2;\n" +
-                        "    specification: LocalH2\n" +
-                        "    {\n" +
-                        "        testDataSetupCSV: 'default\\npersonTable\\nfullName,firmName,addressName\\nP1,F1,A1\\nP2,F2,A2\\nP3,,\\nP4,,A3\\nP5,F1,A1\\n---';\n" +
-                        "    };\n" +
-                        "    auth: DefaultH2;\n" +
-                        "}");
+        String expectedMessage = "{\"completion\":[{\"completion\":\"fullName\",\"display\":\"fullName\"}," +
+                "{\"completion\":\"firmName\",\"display\":\"firmName\"},{\"completion\":\"addressName\"," +
+                "\"display\":\"addressName\"}]}";
+        Iterable<? extends LegendExecutionResult> actual = this.extension.execute(notebook, "notebook", "legend.query.typeahead",
+                executableArgs, Map.of(), notebook.getDocumentState().getGlobalState().cancellationToken("test"));
+        Assertions.assertEquals(1, Iterate.sizeOf(actual));
+        LegendExecutionResult result = actual.iterator().next();
+        Assertions.assertEquals(LegendExecutionResult.Type.SUCCESS, result.getType(), result.getMessage());
+        Assertions.assertEquals(expectedMessage, result.getMessage());
+    }
 
-        stateForTestFactory.newSectionState(gs, "runtime.pure",
-                "###Runtime\n" +
-                        "Runtime  test::h2Runtime\n" +
-                        "{\n" +
-                        "    mappings: [];\n" +
-                        "    connectionStores:\n" +
-                        "    [\n" +
-                        "        test::h2Conn: [ test::h2Store ]\n" +
-                        "    ];\n" +
-                        "}");
+    @Test
+    void cancel()
```

Comment:
@rafaelbey I tried changing this test to use a simple function instead of a function returning a relational (since a function returning a relational won't call `executePlan` with the cancellation token anymore), but the test still fails because it looks like the cancellation token being cancelled isn't causing the command to throw an error. Do you know what might be causing this?